CMakeLists.txt

```diff
@@ -1,187 +1,233 @@
+        " MPLAB_PIC32MEC14XX               - Compiler: MPLAB         Target: PIC32MEC14xx\n"
+        " MPLAB_PIC32MX                    - Compiler: MPLAB         Target: PIC32MX\n"
+        " MPLAB_PIC32MZ                    - Compiler: MPLAB         Target: PIC32MZ\n"
+        " MSVC_MINGW                       - Compiler: MSVC or MinGW Target: x86\n"
+        " OWATCOM_16BIT_DOS_FLSH186        - Compiler: Open Watcom   Target: 16 bit DOS Flsh186\n"
+        " OWATCOM_16BIT_DOS_PC             - Compiler: Open Watcom   Target: 16 bit DOS PC\n"
+        " PARADIGM_TERN_EE_LARGE           - Compiler: Paradigm      Target: Tern EE large\n"
+        " PARADIGM_TERN_EE_SMALL           - Compiler: Paradigm      Target: Tern EE small\n"
+        " RENESAS_RX100                    - Compiler: Renesas       Target: RX100\n"
+        " RENESAS_RX200                    - Compiler: Renesas       Target: RX200\n"
+        " RENESAS_RX600                    - Compiler: Renesas       Target: RX600\n"
+        " RENESAS_RX600_V2                 - Compiler: Renesas       Target: RX600 v2\n"
+        " RENESAS_RX700_V3_DPFPU           - Compiler: Renesas       Target: RX700 v3 with DPFPU\n"
+        " RENESAS_SH2A_FPU                 - Compiler: Renesas       Target: SH2A with FPU\n"
+        " ROWLEY_MSP430F449                - Compiler: Rowley        Target: MSP430F449\n"
+        " RVDS_ARM_CA9                     - Compiler: RVDS          Target: ARM Cortex-A9\n"
+        " RVDS_ARM_CM0                     - Compiler: RVDS          Target: ARM Cortex-M0\n"
+        " RVDS_ARM_CM3                     - Compiler: RVDS          Target: ARM Cortex-M3\n"
+        " RVDS_ARM_CM4_MPU                 - Compiler: RVDS          Target: ARM Cortex-M4 with MPU\n"
+        " RVDS_ARM_CM4F                    - Compiler: RVDS          Target: ARM Cortex-M4 with FPU\n"
+        " RVDS_ARM_CM7                     - Compiler: RVDS          Target: ARM Cortex-M7\n"
+        " RVDS_ARM7_LPC21XX                - Compiler: RVDS          Target: ARM7 LPC21xx\n"
+        " SDCC_CYGNAL                      - Compiler: SDCC          Target: Cygnal\n"
+        " SOFTUNE_MB91460                  - Compiler: Softune       Target: MB91460\n"
+        " SOFTUNE_MB96340                  - Compiler: Softune       Target: MB96340\n"
+        " TASKING_ARM_CM4F                 - Compiler: Tasking       Target: ARM Cortex-M4 with FPU\n"
+        " CDK_THEAD_CK802                  - Compiler: CDK           Target: T-head CK802\n"
+        " XCC_XTENSA                       - Compiler: XCC           Target: Xtensa\n"
+        " WIZC_PIC18                       - Compiler: WizC          Target: PIC18")
+    # Native FREERTOS_PORT for Linux and Windows MINGW builds
+    if(UNIX)
+        message(STATUS " Auto-Detected Unix, setting FREERTOS_PORT=GCC_POSIX")
+        set(FREERTOS_PORT GCC_POSIX CACHE STRING "FreeRTOS port name")
+    elseif(MINGW)
```

Comment:
We often build GCC_POSIX with Mingw, so I would prefer to only include this if using MSVC.

Reply:
This code is only called if FREERTOS_PORT is undefined in the original cmake build step.

If you would like to use GCC_POSIX with Mingw - then that can be done by: `cmake -DFREERTOS_PORT=GCC_POSIX ....` when calling it from MINGW prompt.  And explicitly defining the `FREERTOS_PORT`.

When using `MINGW` as the platform, the most likely candidate is `MSVC_MINGW` - so that is why that was selected when it is not explicitly defined.